src/draw.c

```diff
@@ -281,6 +281,13 @@ static struct dimensions calculate_dimensions(GSList *layouts)
 
         dim.w += 2 * settings.frame_width;
         dim.corner_radius = MIN(dim.corner_radius, dim.h/2);
+
+        /* clamp max width to screen width */
+        const struct screen_info *scr = output->get_active_screen();
+        if (dim.w > scr->w) {
+                dim.w = scr->w;
+        }
```

Comment:
Would it make sense to limit `dim.h` to the screen height as well?

Reply:
Presumably, it wouldn't get used often as the notification height is not really dependent on the content as much as width is.
But technically, I guess it'd make sense in case anyone would want to spread the notification to maximum height for cosmetic reasons.

Reply:
Yeah, for now I think it's better to keep it at just the width.